group-coordinator/src/main/java/org/apache/kafka/coordinator/group/common/TopicIdToPartition.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.coordinator.group.common;
+
+import org.apache.kafka.common.Uuid;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+public class TopicIdToPartition {
```

Comment:
Is this used anywhere? I may have missed it.

Reply:
Used it in the uniform assignor

Reply:
I thought we wanted all the changes in one place including the topicIdPartition structure

Reply:
Right. I would like to have all the changes related to the interface in one PR. However, it must be part of the interface... At the moment, `TopicIdToPartition` is not used. Or, am I missing something?

Reply:
Nit: sorry I keep coming after names, but what does topic id to partition mean? Is this class a `TopicIdPartition` w/o the topic name and augmented by a list of rack ids? If so, the differentiating characteristic of this class seems to be the location of the partition. Should that be reflected in the class name?